api/client/src/main/java/org/projectnessie/client/auth/oauth2/OAuth2Client.java

```diff
@@ -270,10 +278,12 @@ private void maybeWarn(String message, Throwable error) {
     boolean shouldWarn =
         lastWarn == null || Duration.between(lastWarn, now).compareTo(MIN_WARN_INTERVAL) > 0;
     if (shouldWarn) {
+      // defer logging until the client is used to avoid confusing log messages appearing
+      // before the client is actually used
       if (error instanceof HttpClientException) {
-        LOGGER.warn("{}: {}", message, error.toString());
+        used.thenRun(() -> LOGGER.warn("{}: {}", message, error.toString()));
```

Comment:
This could accumulate futures... May be not much, but should we perhaps cancel `used` on close?